kong/core/router.lua

```diff
@@ -821,15 +821,9 @@ function _M.new(apis)
 
   function self.exec(ngx)
     local method      = ngx.req.get_method()
-    local request_uri = ngx.var.request_uri
-    local uri         = request_uri
+    local uri         = ngx.req.uri
```

Comment:
Does this really work? I think we previously had `ngx.var.uri` here. If there is `ngx.req.uri` and that is not badly `normalized` then that could open some possibilities.

Reply:
I tested this with `openresty/1.11.2.4` and there is no such thing as `ngx.req.uri`.

Reply:
Sorry, my mistake.It's `ngx.var.uri`, there is no  `ngx.req.uri`.

Reply:
@samwonwyd Hi, nice to hear back from you! This shouldn't be a concern anyway (we do not want to revert this value to `ngx.var.uri` as both @bungle and I pointed out. Instead, will you be able to provide an implementation based on my above feedback?

Reply:
Ok, I'm trying to write the tests.